packages/compass-components/src/components/generative-ai/generative-ai-input.tsx

```diff
@@ -150,6 +151,10 @@ type GenerativeAIInputProps = {
   isFetching?: boolean;
   placeholder?: string;
   show: boolean;
+  showGuideCue?: boolean;
+  guideCueTitle?: string;
+  guideCueDescription?: string;
+  onHideGuideCue?: () => void;
```

Comment:
The way this is currently implemented, this state doesn't really do anything, you are mixing controlled and uncontrolled components here. If you want component to be controlled, the state of the guide cue should always be read and changed from the outside, not from internal state, if you want it to be uncontrolled, you don't need `onHideGuideCue` and this `showGuideCue` should rather be called something like `showGuideCueOnMount` or `initialShowGuideCue` to indicate that it's not fully controlling the state of the cue component